File: app/src/main/java/ru/ra66it/updaterforspotify/mvp/view/SpotifyView.java
package ru.ra66it.updaterforspotify.mvp.view;


/**
 * Created by 2Rabbit on 06.11.2017.
 */

public interface SpotifyView {

    void showProgress();

    void hideProgress();

    void showErrorSnackbar(int stringId);

    void showNoInternetLayout();

    void hideNoInternetLayout();

    void hideCardView();

    void showCardView();

    void hideFAB();

    void setUpdateImageFAB();

    void setInstallImageFAB();

    void showFAB();

    void setInstalledVersion(String installedVersion);

    void showLayoutCards();

    void hideLayoutCards();

    void setLatestVersionAvailable(String latestVersion);
}

File: app/src/main/java/ru/ra66it/updaterforspotify/presentation/mvp/view/SpotifyView.kt
package ru.ra66it.updaterforspotify.presentation.mvp.view


/**
 * Created by 2Rabbit on 06.11.2017.
 */

interface SpotifyView {

    fun showProgress()

    fun hideProgress()

    fun showErrorSnackbar(message: String)

    fun showNoInternetLayout()

    fun hideNoInternetLayout()

    fun hideCardView()

    fun showCardView()

    fun hideFAB()

    fun setUpdateImageFAB()

    fun setInstallImageFAB()

    fun showFAB()

    fun setInstalledVersion(installedVersion: String)

    fun showLayoutCards()

    fun hideLayoutCards()

    fun setLatestVersionAvailable(latestVersion: String)

    fun requestPermission()

    fun haveSaveFilePermission(): Boolean
}
